subsys/modbus/modbus_serial.c

```diff
@@ -578,6 +579,11 @@ int modbus_serial_init(struct modbus_context *ctx,
 	cfg->uart_buf_ctr = 0;
 	cfg->uart_buf_ptr = &cfg->uart_buf[0];
 
+	/* Drain the fifo */
```

Comment:
This is not acceptable because "[uart_fifo_read() is expected to be called from UART interrupt handler (ISR)"](https://github.com/zephyrproject-rtos/zephyr/blob/dc4eac41980364ddd8e025644596243242688bda/include/zephyr/drivers/uart.h#L765-L775).
Pointless because what it is trying to drain can, if possible, happen immediately after this drain operation. Not necessary as it is treated by RTU protokol.

Reply:
@jfischer-no ok, working on device that switch with a mux the serial connection and can use modbus or uart pipe to read data from two different sensors. Now uart pipe seems that drain the chars on setup. The idea was to avoid to get uart buffer still valid